controllers/tls/mongodb_tls.go

```diff
@@ -96,44 +113,44 @@ func getTLSConfigModification(getUpdateCreator secret.GetUpdateCreator, mdb mdbv
 }
 
 // getCertAndKey will fetch the certificate and key from the user-provided Secret.
-func getCertAndKey(getter secret.Getter, mdb mdbv1.MongoDBCommunity) string {
-	cert, err := secret.ReadKey(getter, tlsSecretCertName, mdb.TLSSecretNamespacedName())
+func getCertAndKey(getter secret.Getter, mdb TLSResource, secretName types.NamespacedName) string {
+	cert, err := secret.ReadKey(getter, mdb.TLSSecretCertName(), secretName)
 	if err != nil {
 		return ""
 	}
 
-	key, err := secret.ReadKey(getter, tlsSecretKeyName, mdb.TLSSecretNamespacedName())
+	key, err := secret.ReadKey(getter, mdb.TLSSecretKeyName(), secretName)
 	if err != nil {
 		return ""
 	}
 
-	return combineCertificateAndKey(cert, key)
+	return CombineCertificateAndKey(cert, key)
 }
 
 // getPem will fetch the pem from the user-provided secret
-func getPem(getter secret.Getter, mdb mdbv1.MongoDBCommunity) string {
-	pem, err := secret.ReadKey(getter, tlsSecretPemName, mdb.TLSSecretNamespacedName())
+func getPem(getter secret.Getter, mdb TLSResource, secretName types.NamespacedName) string {
```

Comment:
We need to add the `secretName` as a parameter here since in enterprise we also support shared clusters, and in that there isn't a single secret name we can get from the resource, so we will pass it to this function